siphon/catalog.py

```diff
@@ -86,20 +93,27 @@ def filter_time_nearest(self, time, regex=None):
             The desired time
         regex : str, optional
             The regular expression to use to extract date/time information from the key. If
-            given, this should contain named groups: 'year', 'month', 'day', 'hour', 'minute',
-            'second', and 'microsecond', as appropriate. When a match is found, any of those
-            groups missing from the pattern will be assigned a value of 0. The default pattern
-            looks for patterns like: 20171118_2356.
+            given, this should contain either
+            1. named groups: 'year', 'month', 'day', 'hour', 'minute', 'second',
+            and 'microsecond', as appropriate. When a match is found, any of those groups
+            missing from the pattern will be assigned a value of 0. The default pattern looks
+            for patterns like: 20171118_2356.
+            or
+            2. a group named 'strptime' (e.g., r'_s(?P<strptime>\d{13})' for GOES-16 data)
```

Comment:
Flake8 isn't happy about the escape `/d` here. I don't know if we need the explicit example in the docstring - if it is, we should add `# noqa:W605`.

Reply:
Kind of curious to see if the raw string fixes this.

I added an example because I like having something to start from so I don't have to look up the regex and strptime docs. Fine with me to match house style, of course.

Reply:
I think it's fine to include, if for no other reason I think 0.001% of our potential audience would otherwise know how to write a regex with a named group without looking at docs. Count myself among the 99.999%.

We shouldn't avoid this in the name of making the tool happy.